src/ldclient_config.erl

```diff
@@ -281,7 +281,23 @@ get_event_schema() ->
 %% @end
 -spec tls_basic_options() -> [ssl:tls_client_option()].
 tls_basic_options() ->
-    tls_basic_options(filelib:is_regular(?HTTP_DEFAULT_LINUX_CASTORE)).
+    case erlang:list_to_integer(erlang:system_info(otp_release)) >= 25 of
+        true -> tls_basic_erlef_options();
+        false -> tls_basic_options(filelib:is_regular(?HTTP_DEFAULT_LINUX_CASTORE))
+    end.
+
+%% @doc Provide basic options for using TLS with the default OTP 25+.
+%% Follows the recommendations from the Erlang Security Working Group.
+%% https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/ssl
+%%
+%% @end
+-spec tls_basic_erlef_options() -> [ssl:tls_client_option()].
+tls_basic_erlef_options() ->
+    CaCerts = public_key:cacerts_get(),
+    [
```

Comment:
If the certs are empty, what is the final behavior? (hopefully that no TLS negotiation succeeds).

Reply:
`:ssl.connect/3` would fail, as the `verify: :verify_peer` mode requires a CA trust store.

If both :cacertfile and :cacerts are blank (empty list/charlist) the options are invalid and `:ssl.connect/3` returns an error without even attempting the connection.